src/core/xml_util/include/openvino/xml_util/xml_deserialize_util.hpp

```diff
@@ -64,81 +28,32 @@ class XmlDeserializer : public ov::AttributeVisitor {
+
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::string>& value) override;
+
+    void on_adapter(const std::string& name, ov::ValueAccessor<bool>& value) override;
     void on_adapter(const std::string& name, ov::ValueAccessor<void>& adapter) override;
 
-    void on_adapter(const std::string& name, ov::ValueAccessor<double>& adapter) override {
-        std::string val;
-        if (!getStrAttribute(m_node.child("data"), name, val))
-            return;
-        adapter.set(stringToType<double>(val));
-    }
-    void on_adapter(const std::string& name, ov::ValueAccessor<int64_t>& adapter) override {
-        std::string val;
-        if (!getStrAttribute(m_node.child("data"), name, val))
-            return;
-        adapter.set(stringToType<int64_t>(val));
-    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<double>& adapter) override;
+    void on_adapter(const std::string& name, ov::ValueAccessor<int64_t>& adapter) override;
 
     void on_adapter(const std::string& name, ov::ValueAccessor<std::shared_ptr<ov::Model>>& adapter) override;
 
-    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<int32_t>>& adapter) override {
-        std::vector<int32_t> value;
-        if (!getParameters<int32_t>(m_node.child("data"), name, value))
-            return;
-        adapter.set(value);
-    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<int32_t>>& adapter) override;
 
-    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<int64_t>>& adapter) override {
-        std::vector<int64_t> value;
-        if (!getParameters<int64_t>(m_node.child("data"), name, value))
-            return;
-        adapter.set(value);
-    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<int64_t>>& adapter) override;
 
-    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<float>>& adapter) override {
-        std::vector<float> value;
-        if (!getParameters<float>(m_node.child("data"), name, value))
-            return;
-        adapter.set(value);
-    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<float>>& adapter) override;
 
-    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<std::string>>& adapter) override {
-        std::vector<std::string> value;
-        if (!getParameters<std::string>(m_node.child("data"), name, value))
-            return;
-        adapter.set(value);
-    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<std::string>>& adapter) override;
+
+protected:
+    virtual ov::Any parse_weightless_cache_attribute(const pugi::xml_node& node) const;
+    virtual void set_constant_num_buffer(ov::AttributeAdapter<std::shared_ptr<ov::AlignedBuffer>>& adapter);
+
+    const pugi::xml_node& get_node() const;
+    const std::shared_ptr<ov::AlignedBuffer>& get_weights() const;
```

Comment:
Implementation is missed

Reply:
I think it can be removed from header now, unless it will be useful in CPU implementation?